seqr/views/apis/data_manager_api.py

```diff
@@ -379,53 +379,62 @@ def update_rna_seq(request, upload_file_id):
         'sampleGuids': [s.guid for s in samples_to_load.keys()],
     })
 
-def load_rna_seq(file_path, user=None, mapping_file=None, ignore_extra_samples=False):
-    sample_id_to_individual_id_mapping = {}
+def _parse_outlier_row(row):
+    yield row['sampleID'], {mapped_key: row[key] for key, mapped_key in RNA_COLUMNS.items()}
+
+def _validate_outlier_header(header):
+    missing_cols = ', '.join([col for col in ['sampleID'] + list(RNA_COLUMNS.keys()) if col not in header])
+    if missing_cols:
+        raise ValueError(f'Invalid file: missing column(s) {missing_cols}')
+
+def load_rna_seq_outlier(file_path, user=None, mapping_file=None, ignore_extra_samples=False):
+    sample_id_to_individual_id_mapping = None
     if mapping_file:
         sample_id_to_individual_id_mapping = load_mapping_file_content(mapping_file)
+    return load_rna_seq(RnaSeqOutlier, file_path, user, sample_id_to_individual_id_mapping, ignore_extra_samples, _parse_outlier_row, _validate_outlier_header)
 
+def load_rna_seq(model_cls, file_path, user, sample_id_to_individual_id_mapping, ignore_extra_samples, parse_row, validate_header):
```

Comment:
Would it be better to put the `load_rna_seq` function to a separated utility file?